### PR TITLE
Add onder-riscv64

### PR DESCRIPTION
**Pull request by furkanonder**

### CODE REVIEW

**Comment by vstinner** [comment]
@zware: Did you send a password to @furkanonder?

**Comment by itamaro** [comment]
> @zware: Did you send a password to @furkanonder?

I see the settings yaml file on the master doesn't have an entry yet for furkanonder. I can set up a password and send him shortly.

File: master/custom/workers.py [hunk 1]
```diff
@@ -314,4 +314,9 @@ def get_workers(settings):
             parallel_tests=4,
             parallel_builders=2,
         ),
+        cpw(
+            name="onder-riscv64",
+            tags=['linux', 'unix', 'ubuntu', 'riscv64'],
+            parallel_tests=4,
+        ),
```
Comment (itamaro) [inline master/custom/workers.py:321]: are all branches working on riscv64? (going back all the way to 3.9)

File: master/custom/builders.py [hunk 2]
```diff
@@ -279,6 +279,9 @@
     # FreBSD x86-64 clang
     # FreeBSD 15 is CURRENT: development branch (at 2023-10-17)
     ("AMD64 FreeBSD15", "opsec-fbsd15", UnixBuild),
+
+    # riscv64 GCC
+    ("riscv64 Ubuntu23", "onder-riscv64", UnixBuild),
```
Comment (vstinner) [inline master/custom/builders.py:284]: This list is for Tier-3, but to get Tier-3, you need one core dev responsible of this platform. I suggest to start with No-Tier for now, it's not a big difference.

**Comment by vstinner** [comment]
Related issue: https://github.com/python/buildmaster-config/issues/473

File: master/custom/workers.py (hunk 1 above)
Comment (furkanonder) [reply master/custom/workers.py]: **3.12**
```sh
== Tests result: SUCCESS ==

12 tests skipped:
    test.test_asyncio.test_windows_events
    test.test_asyncio.test_windows_utils test_devpoll test_ioctl
    test_kqueue test_launcher test_msilib test_perf_profiler
    test_startfile test_winconsoleio test_winreg test_wmi

12 tests skipped (resource denied):
    test_curses test_ossaudiodev test_smtpnet test_socketserver
    test_tix test_tkinter test_ttk test_urllib2net test_urllibnet
    test_winsound test_xmlrpc_net test_zipfile64

464 tests OK.

Total duration: 35 min 22 sec
Total tests: run=41,982 skipped=1,342
Total test files: run=476/488 skipped=12 resource_denied=12
Result: SUCCESS
```

**3.11**
```sh
== Tests result: SUCCESS ==

10 tests skipped:
    test.test_asyncio.test_windows_events
    test.test_asyncio.test_windows_utils test_devpoll test_ioctl
    test_kqueue test_launcher test_msilib test_startfile
    test_winconsoleio test_winreg

12 tests skipped (resource denied):
    test_curses test_ossaudiodev test_smtpnet test_socketserver
    test_tix test_tk test_ttk_guionly test_urllib2net test_urllibnet
    test_winsound test_xmlrpc_net test_zipfile64

461 tests OK.

Total duration: 26 min 52 sec
Total tests: run=40,604 skipped=1,355
Total test files: run=471/483 skipped=10 resource_denied=12
Result: SUCCESS
```

**3.10**
```sh
== Tests result: FAILURE ==

405 tests OK.

1 test failed:
    test_email

19 tests skipped:
    test_curses test_devpoll test_ioctl test_kqueue test_msilib
    test_ossaudiodev test_smtpnet test_socketserver test_startfile
    test_tix test_tk test_ttk_guionly test_urllib2net test_urllibnet
    test_winconsoleio test_winreg test_winsound test_xmlrpc_net
    test_zipfile64

Total duration: 29 min 31 sec
Tests result: FAILURE
```
**3.9**

```sh
== Tests result: FAILURE ==

405 tests OK.

1 test failed:
    test_email

19 tests skipped:
    test_curses test_devpoll test_ioctl test_kqueue test_msilib
    test_ossaudiodev test_smtpnet test_socketserver test_startfile
    test_tix test_tk test_ttk_guionly test_urllib2net test_urllibnet
    test_winconsoleio test_winreg test_winsound test_xmlrpc_net
    test_zipfile64

Total duration: 32 min 59 sec
Tests result: FAILURE
```

There is only one failed test case on branches `3.10` and `3.9`.
```python
test test_email failed -- Traceback (most recent call last):
  File "/home/ubuntu/cpython/Lib/test/support/__init__.py", line 771, in inner
    return func(*args, **kwds)
  File "/home/ubuntu/cpython/Lib/test/test_email/test_utils.py", line 155, in test_variable_tzname
    self.assertEqual(t1.tzname(), 'MSK')
AssertionError: 'Europe' != 'MSK'
- Europe
+ MSK

== Tests result: FAILURE ==

1 test failed:
    test_email

Total duration: 27.9 sec
Tests result: FAILURE
```

Comment (furkanonder) [reply master/custom/workers.py]: `3.9` and `3.10` branches are in [security status](https://devguide.python.org/versions/#supported-versions). Do we need to fix the tests in this case? @itamaro

Comment (itamaro) [reply master/custom/workers.py]: No, excluding those branches is the right thing to do.

**Review by itamaro — APPROVED** [review]
LG!

**Comment by vstinner** [comment]
The worker works as expected and the first build is a success, nice! https://buildbot.python.org/all/#/builders/1379/builds/1

**Comment by furkanonder** [comment]
> @furkanonder: I cannot support such platform unless I have access to it or access to another RISC-V machine. Moreover, I'm not really volunteer to support it :-) But you can ping me if you need help to investigate some issues.
> 
> Congrats for running your worker :-) Long life to this new cool CPU ISA !

Thank you so much for your time and efforts! @vstinner

**Pull request merged**